### PR TITLE
Add fetch test to complement PR #28870

### PR DESCRIPTION
**Pull request by letitz**
This test requires running against Chrome 92. CI is still using Chrome 91, hence it does not yet pass. Splitting this off to its own PR to unblock #28870.

### CODE REVIEW

**Comment by foolip** [comment]
@letitz, https://github.com/web-platform-tests/wpt/pull/29095 unpinned Chrome so we running Chrome 93 actually. (It was 92 briefly.)

**Comment by letitz** [comment]
Ah, ok. I can rebase onto the latest state of the repo once PR #28870 lands.

**Comment by letitz** [comment]
Rebased, CI is running.

**Comment by letitz** [comment]
@foolip @jgraham CI is green, I think this change is ready for review.

**Comment by letitz** [comment]
@foolip @jgraham friendly ping!

**Comment by letitz** [comment]
@foolip @jgraham any chance you could take a look?

File: infrastructure/assumptions/non-local-ports.sub.window.js
Comment (foolip) [inline infrastructure/assumptions/non-local-ports.sub.window.js:106]: This depends on the error message. https://wpt.fyi/results/infrastructure/assumptions/non-local-ports.sub.window.html?label=pr_head&max-count=1&pr=29248 shows Firefox and Safari failing because the fetch succeeded, but if they implement this they'll most likely fail due to not having exactly "Failed to fetch" as the error message.

I'll suggest below what to do instead.

File: infrastructure/assumptions/resources/fetch-and-post-result.html
Comment (foolip) [inline infrastructure/assumptions/resources/fetch-and-post-result.html:5]: In order to assert only that the error was a "TypeError", which is (roughly) what `assert_throws_js` would do, I'd suggest instead posting `error.name` here.

And to make the failure more obvious in the case the fetch succeeds, I'd suggest a more descriptive string. All in all:

```suggestion
        .then(response => {parent.postMessage(`fetch succeeded with status ${response.status}`, "*")})
        .catch(error => {parent.postMessage(`fetch failed with ${error.name}`, "*")});
```

It could be made a bit less verbose, but something like that.

File: infrastructure/assumptions/non-local-ports.sub.window.js
Comment (foolip) [inline infrastructure/assumptions/non-local-ports.sub.window.js:11]: Nit: since I am also suggesting other changes, can I request dropping the `k` prefix here and below? That's familiar to Chromium C++ contributors, but isn't a common style in WPT.

**Review by foolip — CHANGES_REQUESTED** [review]
Thanks for the persistent pings @letitz! Last week I was out, but I have no excuse for the ping before that. I was hoping to speedily merge this now, but spotted an issue with the pass condition that I think will have to be fixed sooner or later, so let's make it now.

File: infrastructure/assumptions/non-local-ports.sub.window.js
Comment (letitz) [reply infrastructure/assumptions/non-local-ports.sub.window.js]: Makes sense. See below.

File: infrastructure/assumptions/resources/fetch-and-post-result.html
Comment (letitz) [reply infrastructure/assumptions/resources/fetch-and-post-result.html]: Great idea, done.

File: infrastructure/assumptions/non-local-ports.sub.window.js
Comment (letitz) [reply infrastructure/assumptions/non-local-ports.sub.window.js]: Done.

**Comment by letitz** [comment]
Thanks @foolip for the review, and no worries. I was not blocked on this, I just wanted to close the loop here.

**Review by foolip — APPROVED** [review]
Thanks @letitz! The failure in https://wpt.fyi/results/infrastructure/assumptions/non-local-ports.sub.window.html?label=pr_head&max-count=1&pr=29248 for Firefox and Safari is very clear now.

**Pull request merged**
